### DISCUSSION TITLE
initial condition settings！！！

### DISCUSSION THREAD

**Discussion by whtwu** [lululxvi/deepxde#932] [Q&A]
Dr Lu, thank you for your extraordinary work:
   my engineering background is to add an initial pulse excitation of 1000N at a certain position of the beam, when using deepxde, I don't know if the Dirac function is used in the initial conditions to represent this pulse excitation, The following is the wrong code I wrote myself, how should the correct code be expressed?
`ic = dde.IC(geomtime, lambda x: 1000*DiracDelta(x[:, 0:1]-0.2), lambda _, on_initial: on_initial)`

**Comment by lululxvi** [comment]
https://github.com/lululxvi/deepxde/issues/200

**Reply by whtwu** [reply]
@lululxvi Thank you very much your reply. I have a couple of questions to help me understand Deepxde. 
The first is that I add a vector not a function to the initial conditions. The code is as follows. Is this possible? 
`initial_values = np.load('initial_values.npy')
#initial_values  = initial_values.flatten()[:, None]
ic = dde.IC(geomtime, lambda x: initial_values, lambda _, on_initial: on_initial)`
And is the num_initial selection random? How do I set up to select points in order to match my initial data set, looking forward to your reply

**Reply by forxltk** [reply]
Seems you should use PointSetBC, not IC.

**Reply by whtwu** [reply]
My engineering background is that there is a pulse excitation on the beam, but I plan to use the data from MATLAB as the initial condition, you mean in the code implementation, it is possible to use PointSetBC instead of IC, right?

**Comment by whtwu** [comment]
I may understand that IC is mostly given by a function. In PointSetBC, a series of points at time 0 along the x direction can be set to achieve the same effect as IC.

**Reply by lululxvi** [reply]
Yes.
